example-nextjs/pages/index.tsx

```diff
@@ -56,6 +59,26 @@ eratosthenes(100)`)
       .then((pyodide) => setPyodide(pyodide))
   }, [])
 
+  function handleEditorDidMount(editor) {
+    editorRef.current = editor
+  }
+
+  const addKeyBinding = (
```

Comment:
I feel I should put this method in utils.
I can pass one more argument `editorRef`,
But then I am skeptical the function having 4 arguments is not really good,

Suggestions plz @jacobjzhang @mcintyre94

Reply:
Two suggestions:

1. We can leverage TypeScript here and create a `CustomKeyBinding` type that houses the four parameters as an object. That way you're only passing in one object in the form of a `CustomKeyBinding` like:

```js
const runCodeBinding: CustomKeyBinding = {
  label: 'run',
  keybinding: CtrlEnter,
  callback: async () => runCode(inputCode, pyodide)
};

addKeyBinding(runCodeBinding);
```

2) I also notice you're invoking this on function parse-- we probably want to use `useEffect` and only on the first instance so that execution is managed by React.

Reply:
So after taking point 1 and 2 it will look something like this.
```typescript
  
  useEffect(() => {
    addKeyBinding(runCodeBinding)
  }, [inputCode])
  
  interface CustomKeyBinding {
    label: string
    keybinding: any
    callback: () => void
    editorRef: any
  }
  const runCodeBinding: CustomKeyBinding = {
    label: 'run',
    keybinding: CtrlEnter,
    callback: async () => runCode(inputCode, pyodide),
    editorRef,
  }

  const addKeyBinding = ({
    label,
    keybinding,
    callback,
    editorRef,
  }: CustomKeyBinding) => {
    editorRef?.current?.addAction({
      id: 'label',
      label,
      keybindings: [keybinding],
      precondition:
        '!suggestWidgetVisible && !markersNavigationVisible && !findWidgetVisible',
      run: callback,
    })
  }
```

Reply:
Though calling `addKeyBinding(runCodeBinding)` in `useEffect` gives one bug that when first-time user keypress ctrl+enter the code does run but when next time he keypress ctrl+enter then the code runs